examples/concurrency/lib/isolate_spawn_syntax.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:convert';
+import 'dart:io';
+import 'dart:isolate';
+
+// #docregion
+class BackgroundWorker {
+  late Isolate _isolate;
+  late SendPort workerIsolateSendPort;
+
+  Future<void> initIsolate() async {
+    // Create a recieve port to recieve messages from spawned isolate
+    final mainIsolateReceivePort = ReceivePort();
+    mainIsolateReceivePort.listen((dynamic message) {
```

Comment:
This comment has really confused me, so I'll try to address different parts in order...

> Consider using forEach instead of listen.
Makes sense

> It's a little worrisome that there is no way to close the port. That means it will keep the current isolate alive forever.
There is a way to close the port, unless I'm misunderstanding something. Toward the bottom of the file, there's a `dispose` method which calls _isolate.kill

> Consider making this an async factory:
> rather than creating an uninitialized object with public uninitialized `late` fields. (That's generally discouraged.)
Makes sense

> Generally, I'm not sure what this is an example of. It's doing intricate setup, then only to do nothing but `print` on the results that come back. That dissonance irks me :)
This example is only showing syntax of how one would set up a long-lived isolate. It purposefully leaves off everything that isn't necessary.

> Personally, I'd probably use `Isolate.run` to start the other isolate, instad of `Isolate.spawn`, 
This example is under the heading "Long-lived isolates" and the file is called "isolate_spawn_syntax". Unless I'm missing something, it would not be appropriate to use Isolate.run here.


If I am missing something, which would most likely be that I don't understand `Isolate.run`, then I'm happy to change this example, but at the moment it feels like from a high-level, the example is "correct" (barring some implementation details that you pointed out).

Reply:
@lrhn I've updated the code to include the async factory, but I'm still wondering if other portions are incorrect